### PR TITLE
Tests/functional tests

### PR DESCRIPTION
**Pull request by rgildein**
- add test bundle
 - add deploymnet test
 - add prometheus2 targets configuration test
 - change converage test, due to need to ignore integration tests

### CODE REVIEW

File: src/charm.py [hunk 1]
```diff
@@ -93,8 +93,7 @@ def private_address(self) -> str:
     def _on_install(self, _):
         """Installation hook that installs prometheus-bind-exporter daemon."""
         self.unit.status = MaintenanceStatus("Installing prometheus-bind-exporter")
-        snap_file = self.model.resources.fetch("prometheus-bind-exporter")
-        subprocess.check_call(["snap", "install", "--dangerous", snap_file])
+        subprocess.check_call(["snap", "install", "prometheus-bind-exporter", "--candidate"])
```
Comment (auria) [inline src/charm.py]: `--candidate` should be removed so stable is used.

File: tests/unit/test_charm.py [hunk 2]
```diff
@@ -109,11 +106,9 @@ def test_private_address(self):
 
     def test_on_install(self):
         """Test install hook."""
-        exp_call = mock.call(["snap", "install", "--dangerous",
-                              "prometheus-bind-exporter.snap"])
+        exp_call = mock.call(["snap", "install", "prometheus-bind-exporter", "--candidate"])
```
Comment (auria) [inline tests/unit/test_charm.py]: same.

**Review by auria — COMMENTED** [review]
Minor change requested. Other than that, +1 (Please also have a look to the feedback in the spec, in case you need to adapt anything in the charm [probably in a different PR]).

**Comment by mkalcok** [comment]
Sorry if this was discussed elsewhere, but why did you switch from installing snap from resource file to downloading directly from snap store? Is this approach acceptable in charms now?

**Comment by rgildein** [comment]
> Sorry if this was discussed elsewhere, but why did you switch from installing snap from resource file to downloading directly from snap store? Is this approach acceptable in charms now?

We discussed it on stand-up I think. In this case it make sense, due:
1. We created that snap and updates of that snap will be rare.
2. When you are using snap resource, you need to maintenance it in snapstore. (update revision if new change happens)
3. User could pass their own snap resource, which could work differently or even not works at all.

^ @auria can correct/supplement me.

**Comment by mkalcok** [comment]
Ok, I understand now that using snapstore instead of resource is suggested approach. Thanks.

**Review by mkalcok — APPROVED** [review]

**Pull request merged**